index/scorch/segment/zap/merge.go

```diff
@@ -649,7 +649,10 @@ func mergeStoredAndRemap(segments []*SegmentBase, drops []*roaring.Bitmap,
 				fieldID := int(fieldsMap[field]) - 1
 				vals[fieldID] = append(vals[fieldID], value)
 				typs[fieldID] = append(typs[fieldID], typ)
-				poss[fieldID] = append(poss[fieldID], pos)
+
+				poss[fieldID] = append(poss[fieldID], make([]uint64, len(pos)))
+				copy(poss[fieldID][len(poss[fieldID])-1], pos)
```

Comment:
this is a bit of a weird formulation of this that feels unnatural, but perhaps logically correct.

another thought that pops up is this might increase memory usage to O(numDocs), and O(numDocs) trips to allocator, so... hmmm.  (meaning, noodling about alternatives)

Reply:
more as a 'note to self' on this, it looks like the VisitDocument() might invoke the visitor callback with a reused pos array, so the callback should copy the values if it wants to hold onto those values...

https://github.com/blevesearch/bleve/blob/master/index/scorch/segment/zap/segment.go#L356

...so the fix here is logically correct, but perhaps more performance is available by copying instead to a giant preallocated buffer

Reply:
one edge case on this... if the pos in the callback has length 0, then we're always allocating something?

that is, make([]uint64, 0) seems to return something non-nil